lib/Twig/Extension/Core.php

```diff
@@ -852,6 +852,41 @@ function twig_in_filter($value, $compare)
     return false;
 }
 
+/** 
+ * Returns a trimmed string.
+ * 
+ * @param string $string
+ * @param string $characterMask
+ * @param string $type
+ *
+ * @return string
+ *
+ * @throws Twig_Error_Runtime When an invalid trimming type is used (not a string or not 'left', 'right' or 'both')
+ */
+function twig_trim_filter($string, $characterMask = null, $type = 'both')
+{
+    if (null === $characterMask) {
+        $characterMask = " \t\n\r\0\x0B";
+    }
+
+    switch ($type) {
+        case 'both':
+            return trim($string, $characterMask);
+        case 'left':
+            return ltrim($string, $characterMask);
+        case 'right':
+            return rtrim($string, $characterMask);
+        default:
+            if (!is_string($type)) {
+                throw new Twig_Error_Runtime(sprintf('Trimming type must be a string but got %s.', gettype($type)));
```

Comment:
I would remove this test a we are not doing that anywhere else and instead, change the condition that follows to: `if (!is_array($type) || !in_array($type, array('both', 'left', 'right'))) {`